test/test_meta.py

```diff
@@ -0,0 +1,837 @@
+                    # instrumentation will see the meta conversions and the
+                    # tests all break so we just exclude this.  In any case
+                    # the to conversion isn't really right anyhow.
+                    miss += 1
+                    return t
+                elif any([
+                    t.device.type in ("lazy", "meta"), t.is_complex(),
+                    torch._C._is_batched(t),
+                ]):
+                    # TODO: this stuff should support storage
+                    # (well, maybe not batched)
+                    hit += 1
+                    return t.to("meta")
+                else:
+                    hit += 1
+                    r = meta_tensor(t)
+                    if type(t) is torch.nn.Parameter:
+                        r = torch.nn.Parameter(r, requires_grad=r.requires_grad)
+                    return r
+            elif torch.overrides.is_tensor_like(t):
+                # Blindly converting tensor subclasses to meta can cause
+                # unpredictable problems; e.g., FX tests will trace meta
+                # tensors into their trace / some subclasses don't correctly
+                # support meta.  Trying to YOLO this is more trouble than it's
+                # worth.
+                miss += 1
+                return t
+            else:
+                # non-Tensor types don't count as hit or miss
+                return t
+
+        do_meta = (
+            func not in meta_exclude_set and not torch.jit.is_tracing() and
+            not isinstance(func, torch.ScriptMethod)
+        )
+
+        if do_meta:
+            try:
+                meta_args = tree_map(to_meta, args)
+                meta_kwargs = tree_map(to_meta, kwargs)
+            except Exception as e:
+                raise RuntimeError(
+                    f"failed to convert args to meta; "
+                    f"originally (*{args}, **{kwargs})") from e
+
+        rs = func(*args, **kwargs)
+
+        # TODO: also handle cases where func raise an exception
+
+        # For now, only attempt if we managed to convert all tensor types
+        # (if any of them failed, we're in a mixed device situation and
+        # this isn't well supported)
+        if do_meta and hit > 0 and miss == 0:
+            try:
+                # suppress warnings
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    meta_rs = func(*meta_args, **meta_kwargs)
+            except Exception as e:
+                suppress = False
+                if isinstance(e, NotImplementedError):
+                    m = RE_NOT_IMPLEMENTED_MSG.search(e.args[0])
+                    if m and m.group(1) not in ("aten::_efficientzerotensor", "aten::view_as_real"):
+                        suppress = True
+                if not suppress:
+                    raise RuntimeError(f"""\
+failed to run: {func}(
```

Comment:
indentation

Reply:
actually unindenting triple blocks like this is idiomatic, because indenting them will result in indentation in the underlying string